### PR TITLE
Noto Naskh Arabic: Version 2.016 added

### PR DESCRIPTION
**Pull request by simoncozens**
c0d2c12: [gftools-packager] Noto Naskh Arabic: Version 2.014 added

* Noto Naskh Arabic Version 2.014 taken from the upstream repo https://github.com/notofonts/arabic.git at commit https://github.com/notofonts/arabic/commit/.

### CODE REVIEW

**Comment by vv-monsalve** [comment]
`qa` diffenator report in this case confirms the ones reported for the `UI` version #5019 

**Marks modified**
marks_modified

**Glyphs Missing**
glyphs_missing

**Comment by vv-monsalve** [comment]
Diffbrowsers images show the marks change

**Windows 10 Chrome 71**
Desktop_Windows_10_chrome_71 0_

**Comment by vv-monsalve** [comment]
Confirm that pushing Urdu figure localization performed in #5025 is included here

**Comment by vv-monsalve** [comment]
Simon had [replied about these issues](https://github.com/google/fonts/pull/4867#issuecomment-1177588698) in the previous #4867 PR for Naskh font, so probably they are good to go.

However, @chrissimpkins said it would be best to confirm if changing the default figures (from Urdu to Farsi) would be safe in some environments that probably would not support the `locl` feature.

**Comment by simoncozens** [comment]
Wait, please, some of those changed marks are new, they look bad, and I don't understand them.

**Comment by chrissimpkins** [comment]
> it would be best to confirm if changing the default figures (from Urdu to Farsi) would be safe in some environments that probably would not support the locl feature.

Discussed during our pipeline meeting this week and this change was approved.

File: ofl/notonaskharabic/METADATA.pb [hunk 1]
```diff
@@ -10,15 +10,21 @@ fonts {
...
+subsets: "latin-ext"
 subsets: "menu"
 axes {
   tag: "wght"
   min_value: 400.0
   max_value: 700.0
 }
+source {
+  repository_url: "https://github.com//notofonts/arabic.git"
+  archive_url: "https://github.com/notofonts/arabic/releases/download/NotoNaskhArabic-v2.013/NotoNaskhArabic-v2.013.zip"
```
Comment (chrissimpkins) [inline ofl/notonaskharabic/METADATA.pb:26]: Title of this PR states v2.014, does this need to be updated?

File: ofl/notonaskharabic/article/ARTICLE.en_us.html [hunk 2]
```diff
@@ -1,34 +0,0 @@
-<p>
```
Comment (chrissimpkins) [inline ofl/notonaskharabic/article/ARTICLE.en_us.html:1]: We need to add article back

File: ofl/notonaskharabic/upstream.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+archive: https://github.com/notofonts/arabic/releases/download/NotoNaskhArabic-v2.013/NotoNaskhArabic-v2.013.zip
```
Comment (chrissimpkins) [inline ofl/notonaskharabic/upstream.yaml:1]: Version here states v2.013 but title of this PR is v2.014. Do we need an update here?

**Review by chrissimpkins — CHANGES_REQUESTED** [review]

File: ofl/notonaskharabic/METADATA.pb (hunk 1 above)
Comment (chrissimpkins) [reply ofl/notonaskharabic/METADATA.pb:26]: Discussed during the pipeline meeting today.  It sounds like it does require an update.

File: ofl/notonaskharabic/upstream.yaml (hunk 3 above)
Comment (chrissimpkins) [reply ofl/notonaskharabic/upstream.yaml:1]: Discussed during the pipeline meeting today.  This does require an update.

**Comment by simoncozens** [comment]
Updated

## Noto Naskh Arabic: Version 2.014 added

---
 bec87e6: [gftools-packager] Noto Naskh Arabic: Version 2.014 added

* Noto Naskh Arabic Version 2.014 taken from the upstream repo https://github.com/notofonts/arabic.git at commit https://github.com/notofonts/arabic/commit/.

File: ofl/notonaskharabic/METADATA.pb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 name: "Noto Naskh Arabic"
 designer: "Google"
 license: "OFL"
-category: "SANS_SERIF"
+category: "SERIF"
```
Comment (chrissimpkins) [inline ofl/notonaskharabic/METADATA.pb:4]: Intentional change?

File: ofl/notonaskharabic/METADATA.pb [hunk 5]
```diff
@@ -1,7 +1,7 @@
 name: "Noto Naskh Arabic"
```
Comment (chrissimpkins) [inline ofl/notonaskharabic/METADATA.pb:1]: Do we want to add a `primary_script` definition?

File: ofl/notonaskharabic/METADATA.pb (hunk 4 above)
Comment (simoncozens) [reply ofl/notonaskharabic/METADATA.pb:4]: Yes, we decided this was better paired with a Serif latin.

Comment (chrissimpkins) [reply ofl/notonaskharabic/METADATA.pb:4]: Ty!

**Review by chrissimpkins — APPROVED** [review]

**Comment by RosaWagner** [comment]
@simoncozens lots of diff in the last version

Desktop_OS_X_High_Sierra_safari_11 1_
glyphs_missing
marks_missing
metrics_modified

**Comment by m4rc1e** [comment]
Diffenator2 report:  [noto_naskh.zip](https://github.com/google/fonts/files/9831740/noto_naskh.zip)

**Comment by simoncozens** [comment]
Mark's diffs look OK to me.

**Comment by chrissimpkins** [comment]
> Mark's diffs look OK to me.

We have a diff in the diffs? :) 

Are we ready to merge?

**Comment by simoncozens** [comment]
I'm not sure. Marc said on the call today that he wanted to point out something about this diffenator2 result, and I don't think I got to hear what it was. @m4rc1e?

**Comment by m4rc1e** [comment]
Here's a few things I spotted which look suspicious.

before:
Screenshot 2022-10-25 at 14 35 21

after:
Screenshot 2022-10-25 at 14 35 46


before:
Screenshot 2022-10-25 at 14 36 20


after:
Screenshot 2022-10-25 at 14 36 40


before:
Screenshot 2022-10-25 at 14 37 57


after:
Screenshot 2022-10-25 at 14 38 40


before:
Screenshot 2022-10-25 at 14 40 03

after:
Screenshot 2022-10-25 at 14 40 44

Otherwise good.

**Comment by simoncozens** [comment]
From previous PR:
> The placement of dots in ڭ appears to be by design, since that's composed as a single glyph.

Mark-to-Mark looks like a problem, though, and I'm not sure about the turned comma.

**Comment by simoncozens** [comment]
OK: The comma issue is not a comma, it is ARABIC THOUSANDS SEPARATOR and the left-facing form is correct. (Input text is wrong.) The mark-to-mark issues were not mark to marks but simply bad anchor placement. I'll fix them and re-release.

**Comment by simoncozens** [comment]
Updated

## Noto Naskh Arabic: Version 2.015 added

---
 ffa21ad: [gftools-packager] Noto Naskh Arabic: Version 2.015 added

* Noto Naskh Arabic Version 2.015 taken from the upstream repo https://github.com/notofonts/arabic.git at commit https://github.com/notofonts/arabic/commit/.

**Comment by m4rc1e** [comment]
Thanks Simon,

Hamza below with Kasra still appears too tight. 

Before:

Screenshot 2022-10-27 at 09 57 19


After:

Screenshot 2022-10-27 at 09 54 52

Imo we need a design phase sign off. At other foundries we'd send these issues back to a designer to fix.

**Comment by simoncozens** [comment]
Urgh, for some reason the release is not what's in the repo. I'm going to pull the release and do it again.

**Comment by simoncozens** [comment]
Updated

## Noto Naskh Arabic: Version 2.015 added

---
 88d68ea: [gftools-packager] Noto Naskh Arabic: Version 2.015 added

* Noto Naskh Arabic Version 2.015 taken from the upstream repo https://github.com/notofonts/arabic.git at commit https://github.com/notofonts/arabic/commit/.

**Review by m4rc1e — APPROVED** [review]
Thanks Simon. LGTM.

I'm still unsure about the triple dots above being so high on kaf forms but if it's intentional as you say then fine.

**Comment by RosaWagner** [comment]
@simoncozens @m4rc1e I don't understand the state of this PR; is it approved? does it need confirmation? Is it unsure?

What about this mark shift? 
Desktop_OS_X_High_Sierra_safari_11 1_

**Comment by simoncozens** [comment]
The character ڭ is composed in the font as a single glyph, i.e. it isn't a random mark positioning problem, but the designer has put those components together in those locations. I agree that the positioning of the three dots looks unusual; I don't know if it's deliberate or not. I'm assuming that it is deliberate.

**Comment by vv-monsalve** [comment]
I finally managed to produce new diffenator2 images where the Arabic words display better. I had doubts indeed the three dots position, but I also have doubts about أيضاًً and حادًّا although by copy-pasting those words here it seems they are ok.

Placing the comparative image in any case here.
NotoNaskh-comparison

**Comment by chrissimpkins** [comment]
Who is the designer?

**Comment by simoncozens** [comment]
Well, Noto is an open source project with many contributors now, so I'm not sure how helpful it is to talk about a single designer. Perhaps we should get into the habit of asking, who made the change and who approved it...? :-)

2.001 had the ng three-dots in the "normal" place. The raised dots were introduced in 2.012, possibly by Nizar Qandah. @nizarsq, could you comment about the position of the three-dots in the ng?

2.001 shape -> 2.012 shape

**Comment by nizarsq** [comment]
@simoncozens I quickly checked my old notes but I didn't find anything related to 06AD that I worked on.

**Comment by chrissimpkins** [comment]
Who made the change and who approved it?

**Comment by simoncozens** [comment]
Well, I think Nirzaq made it and Marek merged it. But this time, Kazunari Tsuboi has fixed it and I have merged it!

**Comment by vv-monsalve** [comment]
Three dots look good. But receiving a new significant change for ٴتىيىس . @simoncozens could you confirm it's ok?

Desktop_Windows_10_chrome_71 0_

**Comment by simoncozens** [comment]
The other expected change is the high hamza glyph. Explained [here](https://github.com/notofonts/arabic/pull/39) . (Maybe I’ll find a way to post the release announcement in the Google PR.)

**Comment by chrissimpkins** [comment]
Have we documented the intentional Urdu ==> Persian default figure change in a release announcement / changelog?

**Comment by simoncozens** [comment]
> Have we documented the intentional Urdu ==> Persian default figure change in a release announcement / changelog?

That change was made (and documented) last year. The recent issue was that it wasn’t also applied to the UI font; but yes, it was documented in the release of the UI font as well.

**Comment by chrissimpkins** [comment]
Thanks Simon

**Comment by RosaWagner** [comment]
@simoncozens No change to  ٴتىيىس  anymore but three dot is shifted again

Desktop_Windows_10_chrome_71 0_

**Comment by simoncozens** [comment]
I think my rebase did something bad, as we're now back to version 2.014.

File: ofl/notonaskharabic/METADATA.pb [hunk 6]
```diff
@@ -22,8 +22,8 @@ axes {
   max_value: 700.0
 }
 source {
-  repository_url: "https://github.com//notofonts/arabic.git"
-  archive_url: "https://github.com/notofonts/arabic/releases/download/NotoNaskhArabic-v2.013/NotoNaskhArabic-v2.013.zip"
+  repository_url: "https://github.com/notofonts/arabic.git"
+  archive_url: "https://github.com/notofonts/arabic/releases/download/NotoNaskhArabic-v2.014/NotoNaskhArabic-v2.014.zip"
```
Comment (chrissimpkins) [inline ofl/notonaskharabic/METADATA.pb:26]: Needs to be updated to v2.016?

File: ofl/notonaskharabic/upstream.yaml [hunk 7]
```diff
@@ -1,5 +1,7 @@
-archive: https://github.com/notofonts/arabic/releases/download/NotoNaskhArabic-v2.013/NotoNaskhArabic-v2.013.zip
+archive: https://github.com/notofonts/arabic/releases/download/NotoNaskhArabic-v2.014/NotoNaskhArabic-v2.014.zip
```
Comment (chrissimpkins) [inline ofl/notonaskharabic/upstream.yaml:1]: Needs to be updated to v2.016?

**Review by chrissimpkins — CHANGES_REQUESTED** [review]

**Comment by RosaWagner** [comment]
I think now it looks like expected and we can merge

Desktop_Windows_10_chrome_71 0_

**Comment by simoncozens** [comment]
I agree.

**Pull request merged**